src/main/java/com/conveyal/taui/controllers/BundleController.java

```diff
@@ -204,11 +219,59 @@ public static Bundle update (Request req, Response res) throws IOException {
         return Persistence.bundles.updateFromJSONRequest(req);
     }
 
-    public static Object getBundle (Request req, Response res) {
-        return Persistence.bundles.findByIdFromRequestIfPermitted(req);
+    public static Bundle getBundle (Request req, Response res) {
+        Bundle bundle = Persistence.bundles.findByIdFromRequestIfPermitted(req);
+
+        // Old bundles did not have these set.
+        try {
+            setBundleServiceDates(bundle);
+        } catch (Exception e) {
+            throw AnalysisServerException.unknown(e);
+        }
+
+        return bundle;
+    }
+
+    public static Collection<Bundle> getBundles (Request req, Response res) {
+        return Persistence.bundles.findPermittedForQuery(req);
+    }
+
+    /**
+     * Old bundles were created without computing the service start and end dates. This finds them and stores them.
+     * @param bundle
+     */
+    public static Bundle setBundleServiceDates (Bundle bundle) throws Exception {
+        if (bundle.status != Bundle.Status.DONE || (bundle.serviceStart != null && bundle.serviceEnd != null)) return bundle;
+
+        // Old bundles were created with computing the service start and end dates
```

Comment:
Not clear what this means: "were created with computing the service start and end dates"

Reply:
Egregious mistype. Meant to say "without" instead of "with". This comment is unnecessary with a more descriptive method comment.